File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        s=list(s)
        t=list(t)

        s.sort()
        t.sort()

        i=0
        j=0

        while i < len(s) and j < len(t):
            if s[i]!=t[j]:
                return False
            i+=1
            j+=1
        
        return True
                    
                

File: Data Structures & Algorithms/car-fleet/submission-3.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int n = position.length;
        int[][] car = new int[n][2];

        for(int i = 0; i<n; i++){
            car[i][0] = position[i];
            car[i][1] = speed[i];
        }

        Arrays.sort(car, (a,b) -> Integer.compare(b[0], a[0]));

        int fleet = 1;
        double prev_time = (double)(target-car[0][0])/car[0][1];

        for(int i = 1; i<n; i++){
            double cur_time = (double)(target-car[i][0])/car[i][1];
            if(cur_time>prev_time){
                fleet++;
                prev_time=cur_time;
            }
        }
        return fleet;
    }
}
